python/pyspark/sql/dataframe.py

```diff
@@ -1557,6 +1557,9 @@ def replace(self, to_replace, value=None, subset=None):
             For example, if `value` is a string, and subset contains a non-string column,
             then the non-string column is simply ignored.
 
+        .. note:: `value` can only be omitted when `to_replace` is a dictionary. Otherwise,
+            it is required.
```

Comment:
Shall we just describe this in `value`'s param doc?

Reply:
Sure.